### PR TITLE
fix: handle url encoded file name in url params for single file serving endpoint

### PR DESCRIPTION
**Pull request by danpiths**
## Summary

URL-encoded path parameters (e.g., file paths containing spaces like
`nested dir/file with spaces.txt`) were not being decoded before use, causing
file serving requests with percent-encoded characters to fail. This PR
introduces a shared helper that centralises path parameter extraction and adds
proper `url.PathUnescape` decoding.

## Changes

- Introduced `decodeStringPathParam` helper that extracts a named path parameter
  from a `fasthttp.RequestCtx`, validates it, and URL-decodes it via
  `url.PathUnescape` before returning it to the caller.
- Replaced duplicated inline path-param extraction logic in `doServeFileContent`
  and `lookupSkillByPathParam` with calls to the new helper.
- Added an integration test
  (`TestSkillsServingGenericFileDownloadDecodesEncodedPathParams`) that creates
  a skill with a file at a path containing spaces, issues a request with a
  percent-encoded URI, and asserts the file content is returned with HTTP 200.

## Type of change

- [x] Bug fix
- [ ] Feature
- [ ] Refactor
- [ ] Documentation
- [ ] Chore/CI

## Affected areas

- [ ] Core (Go)
- [x] Transports (HTTP)
- [ ] Providers/Integrations
- [ ] Plugins
- [ ] UI (React)
- [ ] Docs

## How to test

```sh
go test ./transports/bifrost-http/handlers/...
```

The new test `TestSkillsServingGenericFileDownloadDecodesEncodedPathParams`
exercises the fix end-to-end by serving a file whose path contains spaces via a
percent-encoded URL and asserting the correct body and status code are returned.

## Screenshots/Recordings

N/A

## Breaking changes

- [ ] Yes
- [x] No

## Related issues

N/A

## Security considerations

`url.PathUnescape` is used rather than `url.QueryUnescape` to avoid
misinterpreting `+` as a space in file paths. Decoded paths are validated to be
non-empty before use, preventing empty-string path traversal edge cases.

## Checklist

- [x] I read `docs/contributing/README.md` and followed the guidelines
- [x] I added/updated tests where appropriate
- [x] I updated documentation where needed
- [x] I verified builds succeed (Go and UI)
- [x] I verified the CI pipeline passes locally if applicable

### CODE REVIEW

**Comment by danpiths** [comment]
* **#4546** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/4546?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4539** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/4539?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4521** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/4521?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4464** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/4464?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4450** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/4450?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4433** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/4433?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4428** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/4428?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4427** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/4427?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4426** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/4426?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4422** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/4422?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4487** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/4487?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4495** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/4495?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/maximhq/bifrost/4495?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `dev`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: transports/bifrost-http/handlers/skills_serving_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		t.Fatalf("request failed: %v", err)
+	}
+
+	if resp.StatusCode() != fasthttp.StatusOK {
+		t.Fatalf("status got %d, want %d; body=%s", resp.StatusCode(), fasthttp.StatusOK, string(resp.Body()))
+	}
+	if got := string(resp.Body()); got != string(content) {
+		t.Fatalf("body got %q, want %q", got, string(content))
+	}
+}
```
[the remark this answers is not in the document]
Comment (danpiths) [reply transports/bifrost-http/handlers/skills_serving_test.go:75]: no spaces are allowed for skill names, only alphanumeric and dashes are allowed. please properly go through the codebase before commenting

**Comment by akshaydeo** [comment]
### Merge activity

* **Jun 19, 6:55 AM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/maximhq/bifrost/4495).
* **Jun 19, 6:56 AM UTC**: @akshaydeo merged this pull request with [Graphite](https://app.graphite.com/github/pr/maximhq/bifrost/4495).
